src/addrman.cpp

```diff
@@ -246,12 +246,12 @@ void AddrManImpl::Unserialize(Stream& s_)
 
     uint8_t compat;
     s >> compat;
-    const uint8_t lowest_compatible = compat - INCOMPATIBILITY_BASE;
-    if (lowest_compatible > FILE_FORMAT) {
+    const int lowest_compatible{int{compat} - int{INCOMPATIBILITY_BASE}};
```

Comment:
`int` *is* 32-bit on all of our supported platforms. See compat/assumptions.h. I could change to int32_t, but is there a reason?

Reply:
> is there a reason?

Not any specific reason. This was more of a question out of curiosity rather than a doubt. I have observed that we avoid going with a higher size type for the variable when we can sufficiently work with a smaller size type; that's why I asked the question.

> See compat/assumptions.h.

Let me take a look through it.

Reply:
Reverted hunk